src/modules/tas_optimizer/mod.rs

```diff
@@ -331,10 +333,19 @@ fn optim_init(marker: MainThreadMarker, path: PathBuf, first_frame: usize) {
 
     // TODO: get current parameters.
     let parameters = unsafe {
+        let game_dir = CStr::from_ptr(engine::com_gamedir.get(marker).cast())
+            .to_str()
+            .unwrap();
+
+        let is_paranoia: bool = game_dir.eq("paranoia");
+        let is_cstrike: bool = game_dir.eq("cstrike");
+        let is_czero: bool = game_dir.eq("czero");
+
+        let pmove: *mut playermove_s = *engine::pmove.get(marker);
```

Comment:
Once again we don't want to depend on `pmove` being found. And let's de-duplicate the `sv_maxspeed` part.
```suggestion
        let max_speed = get_cvar_f32(marker, "sv_maxspeed").unwrap_or(320.);
        let client_max_speed = engine::pmove
            .get_opt(marker)
            .map(|pmove| (**pmove).clientmaxspeed)
            .unwrap_or(if is_paranoia { 100. } else { 0. });
```